nntrainer/include/layer.h

```diff
@@ -322,6 +325,29 @@ class Layer {
    */
   bool getFlatten() { return flatten; }
 
+  /**
+   * @brief     Set name of the layer
+   */
+  void setName(std::string name) { this->name = name; }
+
+  /**
+   * @brief     Set name of the layer
+   */
+  std::string getName() const { return name; }
+
+  /**
+   * @brief   Get base name of the layer
+   */
+  virtual std::string getBaseName() = 0;
+
+  /**
+   * @brief     Ensure that layer has a name
+   */
+  void ensureName() {
+    if (name.empty())
+      name = getBaseName() + std::to_string(def_name_count++);
```

Comment:
I think it is better to have prefix on generated name since name sth like "flatten1" could be taken by user.

Reply:
This does not ensure that such a situation won't happen. Rather created a set of names of layer to ensure this.